heartbeat/ipsec

```diff
@@ -0,0 +1,169 @@
+setting up or down a specific tunnel.
+</longdesc>
+<shortdesc lang="en">Handles IPSEC tunnels for VIPs</shortdesc>
+
+<parameters>
+<parameter name="tunnel" unique="1" required="1">
+<longdesc lang="en">
+The name of the tunnel to be monitored.
+</longdesc>
+<shortdesc lang="en">Tunnel name</shortdesc>
+<content type="string" default="" />
+</parameter>
+<parameter name="vip" unique="1" required="1">
+<longdesc lang="en">
+VIP that the tunnel is using.
+</longdesc>
+<shortdesc lang="en">VIP</shortdesc>
+<content type="string" default="" />
+</parameter>
+<parameter name="confdir">
+<longdesc lang="en">
+The directory where the IPSEC tunnel configurations can be found.
+</longdesc>
+<shortdesc lang="en">Tunnel name</shortdesc>
+<content type="string" default="/etc/ipsec.d/" />
+</parameter>
+</parameters>
+
+<actions>
+<action name="start"        timeout="20" />
+<action name="stop"         timeout="20" />
+<action name="monitor"      timeout="20" interval="10" depth="0" />
+<action name="reload"       timeout="20" />
+<action name="meta-data"    timeout="5" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+ipsec_usage() {
+	cat <<END
+usage: $0 {start|stop|monitor|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set. And
+should have a collocation constraint with a VIP associated with the
+tunnel.
+END
+}
+
+ipsec_start() {
+	ipsec auto --add "${OCF_RESKEY_tunnel}"
+	ipsec whack --listen &>> /tmp/ipsec-agent.log
+	local return_code=$?
+	if [ $return_code -eq  1 -o $return_code -eq 10 ]; then
+		ocf_log warn "${OCF_RESOURCE_INSTANCE} : Unable to add tunnel ${OCF_RESKEY_tunnel} with return code ${return_code}"
+		return $OCF_ERR_GENERIC
+	else
+		return $OCF_SUCCESS
+	fi
+}
+
+ipsec_stop() {
+	ipsec auto --down "${OCF_RESKEY_tunnel}"
+	local return_code=$?
+	ocf_log info "${OCF_RESOURCE_INSTANCE} : Put down tunnel ${OCF_RESKEY_tunnel} with return code ${return_code}"
+	return $OCF_SUCCESS
+}
+
+ipsec_monitor() {
+	# Monitor _MUST!_ differentiate correctly between running
+	# (SUCCESS), failed (ERROR) or _cleanly_ stopped (NOT RUNNING).
+	# That is THREE states, not just yes/no.
+
+	ipsec status | grep "$OCF_RESKEY_tunnel" | grep -q unoriented
+	state=$?
+	if [ "$state" == "0" ]; then
+		ip addr show | grep -q "${OCF_RESKEY_vip}"
+		hosting_vip=$?
+		if [ "hosting_vip" == "0" ]; then
+			ocf_log warn "${OCF_RESOURCE_INSTANCE} : tunnel ${OCF_RESKEY_tunnel} is unoriented"
+			return $OCF_ERR_GENERIC
+		else
+			return $OCF_NOT_RUNNING
+		fi
+	else
+		return $OCF_SUCCESS
+	fi
+}
+
+ipsec_validate() {
+	# The tunnel needs to be defined in the configuration
+	cat ${OCF_RESKEY_confdir}/*.conf | grep -q "conn $OCF_RESKEY_tunnel"
+	state=$?
+	if [ "$state" == "0" ]; then
+		return $OCF_SUCCESS
+	else
+		return $OCF_ERR_GENERIC
+	fi
+}
+
+: ${OCF_RESKEY_confdir=${OCF_RESKEY_confdir}}
```

Comment:
This needs to be in the top and use pre-defined _default variable (which makes it set it if it isnt already set).

Example:
https://github.com/ClusterLabs/resource-agents/blob/master/heartbeat/awseip#L42-L49

You can then use the _default variable in the metadata as well instead of hard-coding it.